apps/educational_institutions/models.py

```diff
@@ -0,0 +1,116 @@
+from sqlalchemy import (MetaData, Column, INTEGER, VARCHAR, ForeignKey, Table, JSON)
+from sqlalchemy.orm import relationship
+
+from apps.common.db import Base
+
+metadata_obj = MetaData()
+
+
+class University(Base):
+    __tablename__ = "university"
+
+    university_id = Column(INTEGER, primary_key=True, nullable=False)
+    university_name = Column(VARCHAR(length=255), nullable=False)
+    logo = Column(VARCHAR(length=255))
+    rector_id = Column(INTEGER, ForeignKey("rector.rector_id"))
+
+    rector = relationship("Rector", back_populates="university")
+    faculties = relationship("Faculty", back_populates="university")
+    hostels = relationship("Hostel", back_populates="university")
+    requisites = relationship("Requisites", back_populates="university")
+    user_request_reviews = relationship("UserRequestReview", back_populates="university")
+    user_request = relationship("UserRequest", back_populates="university")
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(university_id="{self.university_id}", university_name="{self.university_name}",' \
+               f'logo="{self.logo}", rector_id="{self.rector_id}")'
+
+
+class Faculty(Base):
+    __tablename__ = 'faculty'
+
+    faculty_id = Column(INTEGER, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False)
+    shortname = Column(VARCHAR(length=20))
+    main_email = Column(VARCHAR(length=50))
+    dekan_id = Column(INTEGER, ForeignKey('dekan.dekan_id'))
+    university_id = Column(INTEGER, ForeignKey("university.university_id"), nullable=False)
+
+    dekan = relationship("Dekan", back_populates="faculties")
+    university = relationship("University", back_populates="faculties")
+    speciality = relationship("Speciality", back_populates="faculties")
+    student = relationship("Student", back_populates="faculties")
+    faculty = relationship("UserFaculty", back_populates="faculties")
+    user_request = relationship("UserRequest", back_populates="faculties")
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(faculty_id="{self.faculty_id}", name="{self.name}", shortname="{self.shortname}", ' \
+               f'main_email="{self.main_email}", dekan_id="{self.dekan_id}", university_id="{self.university_id}")'
+
+
+class Speciality(Base):
+    __tablename__ = 'speciality'
+
+    speciality_id = Column(INTEGER, primary_key=True, nullable=False)
+    code = Column(INTEGER, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False)
+    faculty_id = Column(INTEGER, ForeignKey("faculty.faculty_id"), nullable=False)
+
+    faculties = relationship("Faculty", back_populates="speciality")
+    student = relationship("Student", back_populates="specialties")
```

Comment:
Should it be `specialty`, because:
The student has only 1 specialty
The student related to only 1 faculty
The faculty has many specialties

@anton-serozhechkin could you confirm that?